### PR TITLE
Fix #2692

### LINKED ISSUE #2692: XYZWriter ignores remark argument
## Expected behavior ##

The XYZWriter writes the `remark` argument in the comment/molecule name line between the number of atoms and the coordinate body.

https://github.com/MDAnalysis/mdanalysis/blob/b341090aaa0bace9b946316ebf1e5bf965b026da/package/MDAnalysis/coordinates/XYZ.py#L147-L149

## Actual behavior ##

`XYZWriter.remark` is set but nothing is done with this attribute. Instead, it writes the frame number in that line; there is no way to turn this off.

https://github.com/MDAnalysis/mdanalysis/blob/b341090aaa0bace9b946316ebf1e5bf965b026da/package/MDAnalysis/coordinates/XYZ.py#L263

## Code to reproduce the behavior ##

[From the mailing list issue](https://groups.google.com/forum/#!msg/mdnalysis-discussion/QqBapP48wKg/EPkelhGRAQAJ) -- 

``` python
import MDAnalysis as mda

my_in_xyz="/path/to/input.xyz"
my_out_xyz="/path/to/output.xyz"

system = mda.Universe(my_in_xyz, format="XYZ", dt=1.0, in_memory=True)
system.universe.trajectory[-1]
system.atoms.write(my_out_xyz, remark='')

```

The written file looks like

```
12345
frame 3
       N    -0.00000   -1.00000   -2.00000
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
- Which version of Python (`python -V`)?
- Which operating system?

### PR DESCRIPTION
**Pull request by RMeli**
Fixes #2692

Changes made in this Pull Request:
 - Use user-provided `remark` in `XYZWriter`

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by lilyminium** [comment]
Thanks for working on this, @RMeli! I mentioned earlier that I thought something similar to seaborn's [FacetGrid.set_titles](https://kite.com/python/docs/seaborn.axisgrid.FacetGrid.set_titles) function could be a way to address this -- Seaborn makes the variables `col_name` and `row_name` available for string templating, so you can label your grids with custom titles by passing `g.set_titles("My grid for {col_name}")`. That could be one way to have `remark` be flexible enough to be either about frames, times, or molecule names.

**Comment by RMeli** [comment]
Thanks @lilyminium , I'll have a look. I messed up branches in this PR, so I'll keep it on hold until #2685 is merged.

File: testsuite/MDAnalysisTests/coordinates/test_xyz.py [hunk 1]
```diff
@@ -106,6 +107,30 @@ def test_no_conversion(self, ref, reader, tmpdir):
...
+            2 * ["Curstom Remark"],
+            2 * [""],
+            [None, "Written by MDAnalysis XYZWriter (release {0}) | Frame 0".format(__version__)],
+        ]
+    )
+    def test_remark(self, remarkout, remarkin, ref, tmpdir):
+        """
+        Test remark handling.
+        """
+
```
Comment (lilyminium) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py:121]: Nitpicky, but this docstring and empty line seem redundant

File: testsuite/MDAnalysisTests/coordinates/test_xyz.py [hunk 2]
```diff
@@ -106,6 +107,30 @@ def test_no_conversion(self, ref, reader, tmpdir):
...
+        u = mda.Universe(ref.topology, ref.trajectory)
+
```
Comment (lilyminium) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py:123]: this blank line too

File: package/MDAnalysis/coordinates/XYZ.py [hunk 3]
```diff
@@ -259,8 +267,20 @@ def write_next_timestep(self, ts=None):
...
 
+        # Write number of atoms
         self._xyz.write("{0:d}\n".format(ts.n_atoms))
-        self._xyz.write("frame {0}\n".format(ts.frame))
+
+        # Write remark
+        if self.remark is None:
+            remark = "Written by MDAnalysis {0} (release {1})".format(
+                self.__class__.__name__, __version__
+                ) + " | Frame {0}\n".format(ts.frame)
```
Comment (lilyminium) [inline package/MDAnalysis/coordinates/XYZ.py:277]: Might people who write out .xyz trajectories be caught off-guard by this remark if they used to rely on reading the frame? `Frame {0} | Written by ...` might work better here -- and the more dynamic information is then shown first.

Comment (RMeli) [reply package/MDAnalysis/coordinates/XYZ.py]: Good point! =)

File: testsuite/MDAnalysisTests/coordinates/test_xyz.py [hunk 4]
```diff
@@ -106,6 +107,26 @@ def test_no_conversion(self, ref, reader, tmpdir):
...
+    @pytest.mark.parametrize("remarkout, remarkin", 
+        [   
+            2 * ["Curstom Remark"],
+            2 * [""],
+            [None, "frame 0 | Written by MDAnalysis XYZWriter (release {0})".format(__version__)],
+        ]
+    )
+    def test_remark(self, remarkout, remarkin, ref, tmpdir):
+        u = mda.Universe(ref.topology, ref.trajectory)
+
```
Comment (lilyminium) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py:119]:
```suggestion
```

Comment (RMeli) [reply testsuite/MDAnalysisTests/coordinates/test_xyz.py:119]: I love blank lines! ;)

**Comment by RMeli** [comment]
I added your suggestion and solved the conflicts. Should be good to go.

@lilyminium I'll have a look at your suggestion for enhanced string formatting once 1.0 is out.

**Comment by lilyminium** [comment]
Sounds good :-)

**Pull request merged**